db/view/view.cc

```diff
@@ -2693,12 +2693,94 @@ static future<> flush_base(lw_shared_ptr<replica::column_family> base, abort_sou
     }).discard_result();
 }
 
+future<> view_builder::update_base_schema(const replica::column_family& base) {
+    auto base_schema = base.schema();
+    auto& views = base.views();
+    if (views.empty()) {
+        co_return;
+    }
+    auto step_it = _base_to_build_step.find(base_schema->id());
+    if (step_it == _base_to_build_step.end()) {
+        co_return;// In case all the views for this CF have finished building already.
+    }
+    if (step_it->second.base_schema->version() == base_schema->version()) {
+        co_return;// Base schema was already updated, probably while creating an index.
+    }
+    step_it->second.base_schema = base_schema;
+    // Update view_ptrs in all build statuses with new view_ptr from the base table.
+    // If some were dropped, stop them as well, so we won't continue building them,
+    // especially as their schema is not updated with the new base schema.
+    // If some were added, we can handle them later, in on_create_view.
+    std::vector<view_ptr> views_to_remove;
+    for (auto& status : step_it->second.build_status) {
+        auto view_it = std::ranges::find_if(views, [&status] (const view_ptr& v) {
+            return v->id() == status.view->id();
+        });
+        if (view_it == views.end()) {
+            views_to_remove.push_back(status.view);
+        } else {
+            // This update will be performed again in on_update_view, but it will set the same
+            // view_ptr, and it's still needed there because not all view schema updates are
+            // caused by base schema updates.
+            status.view = *view_it;
+        }
+    }
+    for (auto& view : views_to_remove) {
+        co_await drop_view(base_schema->ks_name(), view->cf_name(), step_it->second);
+    }
+}
+
+future<> view_builder::drop_view(const sstring& ks_name, const sstring& view_name, build_step& step) {
+    // The view is absent from the database at this point, so find it by brute force.
+    for (auto it = step.build_status.begin(); it != step.build_status.end(); ++it) {
+        if (it->view->cf_name() == view_name) {
+            vlogger.info0("Stopping to build view {}.{}", ks_name, view_name);
+            _built_views.erase(it->view->id());
+            step.build_status.erase(it);
+            if (this_shard_id() != 0) {
+                // Shard 0 can't remove the entry in the build progress system table on behalf of the
+                // current shard, since shard 0 may have already processed the notification, and this
+                // shard may since have updated the system table if the drop happened concurrently
+                // with the build.
```

Comment:
I didn't understand this comment. Is it safe to call this function from shard != 0?